snap/snapcraft.yaml

```diff
@@ -136,27 +133,51 @@ parts:
       - PIP_NO_BINARY: ":all:"
       - CRYPTOGRAPHY_OPENSSL_NO_LEGACY: "true"
       - UV_FROZEN: "true"
-      - UV_PYTHON: /usr/bin/python3
+      - MAKEOPTS: -j$(nproc --all)
+      - UV_PYTHON_DOWNLOADS: never
+      - UV_COMPILE_BYTECODE: "true"
+      - CLICOLOR_FORCE: "true"
+      - UV_PROJECT_ENVIRONMENT: $CRAFT_PART_INSTALL
     build-attributes:
       - enable-patchelf
     override-pull: |
-      craftctl default
+      # Need the full path for craftctl because we stage our own craftctl to the path.
+      /snap/snapcraft/current/libexec/snapcraft/craftctl default
       # Ensure we don't have a dubious ownership error from git with a remote build.
       git config --global --add safe.directory $CRAFT_PART_SRC
-      # set the version
-      version="$(python3 setup.py --version)"
-      craftctl set version="${version}"
-      [ -n "$(echo $version | grep "post")" ] && grade=devel || grade=stable
-      craftctl set grade="${grade}"
       # Set up the rust compiler
       rustup default 1.79
     override-build: |
-      uv export --no-dev --no-emit-workspace --output-file $CRAFT_PART_SRC/requirements.txt
-      craftctl default
-      cp -v completion.bash ../install
-      sed -i -e '1 s|^#!/.*|#!/snap/charmcraft/current/bin/python -E|' $CRAFT_PART_INSTALL/bin/craftctl
-    organize:
-      bin/craftctl: libexec/charmcraft/craftctl
+      uv sync --no-dev --no-editable --extra apt-jammy
+      cp -v completion.bash $CRAFT_PART_INSTALL
+
+      mkdir -p $CRAFT_PART_INSTALL/libexec/charmcraft
+      ln -s ../../bin/craftctl $CRAFT_PART_INSTALL/libexec/charmcraft/craftctl
+
+      version="$($CRAFT_PART_INSTALL/bin/charmcraft --version | cut -d' ' -f2)"
+      /snap/snapcraft/current/libexec/snapcraft/craftctl set version="${version}"
+      [ -n "$(echo $version | grep "post")" ] && grade=devel || grade=stable
+      /snap/snapcraft/current/libexec/snapcraft/craftctl set grade="${grade}"
+
+      rm $CRAFT_PART_INSTALL/bin/python*
+      rm $CRAFT_PART_INSTALL/lib64
+      ln -s ../usr/bin/python3 $CRAFT_PART_INSTALL/bin/python3
+      ln -s python3 $CRAFT_PART_INSTALL/bin/python
+    stage:
+      - -bin/activate*
+      - -bin/chardetect
+      - -bin/craft-providers
+      - -bin/distro
+      - -bin/jsonschema
+      - -bin/deactivate*
+      - -bin/keyring
+      - -bin/normalizer
+      - -bin/pip*
+      - -bin/pydoc*
+      - -bin/snap-helpers
+      - -bin/tabulate
+    # organize:
+    #   bin/craftctl: libexec/charmcraft/craftctl
```

Comment:
Did this stop working?

Reply:
The shebang changed because we're in a relocatable venv, and it was easier to just symlink in `override-build`. Should I not do that (does it need not not be available in the default PATH?)

Reply:
I changed it so I'm using organize again and just rewriting the shebang.